### DISCUSSION TITLE
Easier to Access Metadata in Search Results

### DISCUSSION THREAD

**Discussion by ghost** [microsoft-search/pnp-modern-search#3012] [Ideas]
**Describe the solution you'd like**
Our users have a need to be able to directly edit metadata of a document within the context of search results. We currently do this with SharePoint Server 2016 via a custom search results display template by adding a link to the document preview panel which opens the standard edit metadata view in a modal.

With the PnP Search results and modern, the edit metadata is well hidden under the "I" icon and does not expand to show all of the columns making it very difficult to view and edit the metadata.

Our desire would be to have an edit metadata experience in a slide out panel from the right side of the page, identical to the experience for editing metadata in a document library.

**Describe alternatives you've considered**
We have a short term solution of providing a link to the item which takes the user to the classic edit metadata experience in a new tab, but this is suboptimal as it takes them out of the context of the search results.

**Comment by wobba** [comment]
@tseward I think this makes sense as a custom web component. We'd be happy to host it, but not sure it should be in the OOB solution. Also, for multi-geo environments we would have to use the Graph API to ensure we have a valid token.
